entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -89,14 +91,29 @@ public void execute(EntityQueryRequest request, StreamObserver<ResultSetChunk> r
         .convertToEDSQuery(request, attrNameToEDSAttrMap.get(request.getEntityType()));
     Iterator<Document> documentIterator = entitiesCollection.search(
         DocStoreConverter.transform(tenantId.get(), query));
-    List<Entity> entities = convertDocsToEntities(documentIterator);
-
-    //Build result
-    //TODO : chunk response. For now sending everything in one chunk
-    responseObserver.onNext(convertEntitiesToResultSetChunk(
-        entities,
-        request.getSelectionList(),
-        attrNameToEDSAttrMap.get(request.getEntityType())));
+
+    boolean isFirstEntity = true;
+    while (documentIterator.hasNext()) {
+      Optional<Entity> entity = DOCUMENT_PARSER.parseOrLog(documentIterator.next(), Entity.newBuilder());
```

Comment:
Earlier, we are using - `DocStoreJsonFormat` as part of https://github.com/hypertrace/entity-service/blob/4025041f432818bdbfb0b6123cdd2fa7bedeadff/entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java#L103

I think, this function would have been reused from `convertDocsToEntities` to `convertDocsToEntity`.
```
private Entity convertDocsToEntity(Document document) {
      Entity.Builder builder = Entity.newBuilder();
      try {
        PARSER.merge(next.toJson(), builder);
      } catch (InvalidProtocolBufferException e) {
        LOG.error("Could not deserialize the document into an entity.", e);
      }
    }
    return builder.build();
  }
```

Reply:
I am instead using `DocumentParser#parseOrLog`
Infact this method `convertDocsToEntities` should also be updated to use `DocumentParser#parseOrLog`. I will do it in the next pr.

Reply:
I see the reason you converted `DocumentParser#parseOrLog` to the public. But, what was the difference, and the reason you choose to move to use this function?

Reply:
I see it is internally using the same PARSER.merge method.